File: python/m1/m1.2_scratch_agent.py
from deepagents import create_deep_agent

from models import model
from dotenv import load_dotenv

load_dotenv()

agent = create_deep_agent(model=model)

result = agent.invoke({"messages": [{"role": "user", "content": "What is an LLM?"}]})

print(result["messages"][-1].content)

File: typescript/m1/m1.2_scratch_agent.ts
import { createDeepAgent } from "deepagents";
import { HumanMessage } from "@langchain/core/messages";

import { model } from "../models.js";

const agent = createDeepAgent({ model });

const result = await agent.invoke({"messages": [{"role": "user", "content": "What is an LLM?"}]});

console.log(result.messages[result.messages.length - 1].content);
